### ISSUE TITLE
Add simple Request class to Response

### ISSUE THREAD

**Issue by justmobilize** [adafruit/Adafruit_CircuitPython_Requests#187]
I was playing with [ssl: work on anything implementing the socket protocol](https://github.com/adafruit/circuitpython/pull/8954) and testing the quote endpoint on Adafruit (which I know redirects to https) and realized there isn't a way to find that out.

I would like to add a light-weight `Request` class that returns the `method`, `url` and `headers`, like in CPython:

<img width="955" alt="image" src="https://github.com/adafruit/Adafruit_CircuitPython_Requests/assets/1547881/e6e0e7b6-372e-4f32-8908-234f047c913c">

_Here you can see it moved to https_

**Comment by justmobilize** [comment]
@FoamyGuy do you think this would be a good add?

**Comment by FoamyGuy** [comment]
Yeah, It seems like nice functionality to have to me, as long as we stick to a subset of cpython requests behavior and API.
